File: CodingSamples/DotNet/Platform/AsyncParallelTest/DemoApp/Program1.cs
class Program
{
    class Computation
    {
        public long Compute(int first, int last)
        {
            return Enumerable.Range(first, last - first + 1)
                            .Select(Worker.DoWork)
                            .Sum();
        }
    }

    private static void DoComputation(int limit)
    {
        Console.Write("Computing...");
        var w = new System.Diagnostics.Stopwatch();
        var c = new Computation();
        w.Start();
        long r = c.Compute(1, limit);
        w.Stop();
        Console.WriteLine("Done!");
        Console.WriteLine("Result = {0} computed in {1:0.000} seconds.", r, w.Elapsed.TotalSeconds);
    }

    public static void Main(string[] args)
    {
        int n = args.Length > 0 ? int.Parse(args[0]) : 10;
        DoComputation(n);        
    }
}

File: CodingSamples/Java/Platform/AsyncParallelTest/Program1.java
import java.util.stream.*;

class Program {

	static class Computation {
	
		public long compute(int first, int last) {
			return IntStream.range(first, last + 1)
						.mapToLong(Worker::doWork)
						.sum();
		}
	
	}


	private static void doComputation(int limit) {
		System.out.print("Computing...");
		var c = new Computation();
		long t1 = System.nanoTime();
		long r = c.compute(1, limit);
		long t2 = System.nanoTime();
		System.out.println("Done!");
		System.out.printf("Result is %d, computed in %.3f seconds.%n", r, (t2 - t1) / 1e9);
	}

	public static void main(String[] args) throws Exception {
		int n = Integer.parseInt(args[0]);
		doComputation(n);
	}
}

